R/ChromBackend.R

```diff
@@ -796,5 +828,87 @@ setMethod("split", "ChromBackend", function(x, f, drop = FALSE, ...) {
 ################################################################################
 ## Filter functions
 
-## TODO:
-## - generic filterChromData and filterPeaksData
+#' @exportMethod filterChromData
+#'
+#' @note
+#' Filter chromData removes chromatograms that do not match the filter criteria.
+#' FilterPeaks Data does not, only some data points are removed.
+#'
+#' can i make it possible that it can filter both character and numeric columns?
```

Comment:
agree. it makes more sense with numerical values - especially for the `filterChromData()` it would be easy to do that with *standard* R conditions. The advantage of the `filterChromData()` function is that we can maybe have optimized versions for specific backends. Example: a SQL backend might implement the filter function such that the filter is done as an SQL statement, so the database takes care of filtering, and no data needs to be loaded/processed in R.